torch/testing/_internal/distributed/rpc_utils.py

```diff
@@ -49,8 +52,38 @@
 # the agent, which then gets mixed-in with each test suite and each mp method.
 
 
+class RpcMultiProcessTestCase(MultiProcessTestCase):
+    def setUp(self):
+        super().setUp()
+        self.output_dir = tempfile.mkdtemp()
+        self.subprocess_init_data["output_dir"] = self.output_dir
+
+    def subprocess_init(self, data):
+        with open(os.path.join(data["output_dir"], f"rank_{self.rank}_stdout"), "wt") as f:
```

Comment:
In general, it looks like we redirect stdout and err to different files, and then undo that redirection on a failure to print out the errors. Would a context manager similar to the one the one defined here: https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/distributed/distributed_test.py#L136 work for this use case? It might help reduce some of the file handling logic.